students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+
+
+def populateteamdict():
+    # this function creates a dict where the key is
+    # the team number (string) and the value is the team object.
+    for i in range(0, len(teamlist)):
+        teamdict.update({teamlist[i]: teamlist2[i]})
+
+
+def fillteamstats():
+    # this function pulls team data from the matchlist data
+    for i in teamdict:
+        teamdict[i].fillmatches()
+        teamdict[i].fillmatchwins()
+        teamdict[i].fillgearsdelivered()
+        teamdict[i].fillkpa()
+        teamdict[i].fillpilot()
+        teamdict[i].fillalliances()
+        teamdict[i].fillfieldpos()
+        teamdict[i].fillautogears()
+        teamdict[i].fillautokpa()
+        teamdict[i].fillclimb()
+
+
+def createteamlist():
+    # This very Janky function creates a list of team numbers, a matching list of team objects and calls the teamdict function
+    # I really dont like this function it is not really necessary and poorly concieved but I could not get the "elegant" way
+    # to work right and I needed something that worked in this case.
+    for i in range(0,len(matchlist)):
+        testteamnum = matchlist[i].team
+        if testteamnum not in teamlist:
+            teamlist.append(testteamnum)
+    for j in range(0,len(teamlist)):    #I THINK THIS IS SUPER JANKY BUT I COULD NOT GET A DIRECT CLASS LIST CREATION TO WORK! :(
+        teamlist2.append(Team(teamlist[j]))
+    populateteamdict()
+    fillteamstats()
+
+def readmatchfile():
+    # this function imports a match file CSV
+    # in the future I would like this to be done in a database.
+    with open('MatchLedger.csv', 'r') as inputfile:
+        csvreader = csv.reader(inputfile)
+        next(inputfile)
+        for line in csvreader:
+            matchlist.append(Match(line[0], line[1]))
+            for i in range(0, len(matchlist)):
+                if matchlist[i].matchnumber == line[0] and matchlist[i].team == line[1]:    #why did I do this?  To check for quality?
+                    matchlist[i].alliance = line[2]
+                    matchlist[i].fieldpos = line[3]
+                    matchlist[i].win = line[4]
+                    matchlist[i].pilot = line[5]
+                    matchlist[i].gears = line[6]
+                    matchlist[i].autogears = line[7]
+                    matchlist[i].kpa = line[8]
+                    matchlist[i].autokpa = line[9]
+                    matchlist[i].climb = line[10]
+                    matchlist[i].penalties = line[11]
+
+    print()
+    print()
+    print("File has been read.")
+    print()
+    print()
+    createteamlist()
+    primaryUI()
+
+
+def outputmatchdata(matchnumber):
+    for i in range(0, len(matchlist)):
+        if matchlist[i].matchnumber == matchnumber:
+            print("")
+
+
+def produceinsights(teamnum):
+    # This function calls multiple functions to provide a report
+    # containing statistical outcomes that are applicable
+    # to the specified team.
+    pass
+
+
+def doesteamexist(team):
```

Comment:
could you make this simply:

`team in teamlist`

if the types are right -- and comparison of the Team objects works. (define `__eq__`)